webextension/background.js

```diff
@@ -6,14 +6,136 @@
 
 /* globals browser */
 
-const gURLThankYou = "https://mozilla.github.io/webcompat-blipz-experiment/thanks.html";
+let gCurrentlyPromptingTabId;
+
+const Config = (function() {
+  class Config {
+    constructor() {
+      this._neverShowAgain = false;
+      this._skipPrivateBrowsingTabs = true;
+      this._lastPromptTime = 0;
+      this._domainsToCheck = {
+        "accounts.google.com": 0,
+        "amazon.com": 0,
+        "amazon.in": 0,
+        "bing.com": 0,
+        "docs.google.com": 0,
+        "drive.google.com": 0,
+        "facebook.com": 0,
+        "flipkart.com": 0,
+        "github.com": 0,
+        "google.co.in": 0,
+        "google.com": 0,
+        "inbox.google.com": 0,
+        "instagram.com": 0,
+        "linkedin.com": 0,
+        "mail.google.com": 0,
+        "netflix.com": 0,
+        "pandora.com": 0,
+        "play.google.com": 0,
+        "reddit.com": 0,
+        "soundcloud.com": 0,
+        "theverge.com": 0,
+        "twitch.tv": 0,
+        "twitter.com": 0,
+        "web.whatsapp.com": 0,
+        "youtube.com": 0,
+      };
+    }
 
-const gMinimumFrequencyBeforeRePrompting = 1000 * 20; // 20 seconds (for testing)
-const gSkipPrivateBrowsingTabs = true;
+    load() {
+      // TBD
+      return Promise.resolve();
+    }
 
-const gDomainCheckTimestamps = {};
+    save(options) {
+      // TBD
+      console.info("Saving options", options);
+    }
 
-let gNeverShowAgain = false;
+    onUserPrompted(domain) {
+      const now = Date.now();
+      this._lastPromptTime = now;
```

Comment:
Are we going to persist this as a pref, or some other way?

Reply:
I think that's what you mean by

> have deferred the actual implementation of about:config stuff to a future issue.

but just checking.

Reply:
Tom answered this in a bug I hadn't read yet: https://github.com/mozilla/webcompat-blipz-experiment/issues/40#issue-321996151